metadata/src/main/java/org/apache/kafka/controller/BrokerHeartbeatManager.java

```diff
@@ -414,9 +421,11 @@ void updateControlledShutdownOffset(int brokerId, long controlledShutDownOffset)
             throw new RuntimeException("Fenced brokers cannot enter controlled shutdown.");
         }
         active.remove(broker);
-        broker.controlledShutDownOffset = controlledShutDownOffset;
-        log.debug("Updated the controlled shutdown offset for broker {} to {}.",
-            brokerId, controlledShutDownOffset);
+        if (broker.controlledShutDownOffset < 0) {
```

Comment:
Perhaps we should call this `maybeUpdateControlledShutdownOffset` since the update may or may not happen. We should also update the javadoc to describe the change in behavior.